src/Livewire/Portal/Shop/ProductDetail.php

```diff
@@ -56,13 +75,38 @@ public function selectOption(ProductOption $option): void
         }
 
         $this->fillProductForm($product);
+        $this->variant = $product->id;
+    }
+
+    public function downloadMedia(array|int $media, ?string $collectionName = null): void
+    {
+        $media = app(MediaModel::class)->query()->whereIntegerInRaw('id', Arr::wrap($media))->get();
+
+        count($media) > 1
+            ? $this->redirectRoute(
+                'portal.media.download-multiple',
+                [
+                    'ids' => $media->pluck('id')->implode(','),
+                    'filename' => $this->productForm->name . ' - ' . $collectionName,
```

Comment:
only append collectionName if not null, otherwise the filename ends with ' - '

Reply:
√